### PR TITLE
Expose Factory Methods for Console Sink

### PR DESCRIPTION
**Pull request by Mike-EEE**
**What issue does this PR address?**
#36 

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit follows our [guidelines](https://github.com/serilog/serilog-sinks-console/blob/dev/CONTRIBUTING.md)
- [x] Unit Tests for the changes have been added (for bug fixes / features)

I am not sure of the best way to create a useful unit test for this, except to show that the instances instantiate correctly.  Very much open to feedback here!

**Other information**:
*Please list any other relevant information here*
This is my first PR.  Be easy on me. 😆

### CODE REVIEW

**Comment by nblumhardt** [comment]
Hi Mike! Thanks for the PR. Just reading the issue description, I'm not confident that I understand how this all fits together. `Console()` already accepts a custom `ITextFormatter` that you can decorate - but of course, the use of a formatter sidesteps all of the coloring features etc. that make this sink "interesting". Is that what you're aiming for? The loss of coloring could be a bit of a let-down.

If you just want to to wrap the sink, i.e. with a decorator, check out `LoggerSinkConfiguration.Wrap()` - there should be some example usage in the Serilog unit tests and/or the original PR that shows how this is done.

HTH!

**Comment by Mike-EEE** [comment]
Hi @nblumhardt thanks for letting me know about `Wrap` and for your thoughts!  The primary issue here is that both ConsoleSink and its subsequent `ITextFormatter` (`OutputTemplateRenderer`) are internal, so I as an external consumer cannot new them up, even for use for `Wrap`, which is exactly what am looking to do, actually.  I created the factory methods as I was assuming you were wanting to keep these components internal for whatever reason.  Simply marking them public would be another viable path forward.

To ensure clarity here, the use of `Wrap` would require instantiation as well, so that cannot be utilized here, I am afraid.  Not that I can see at least from checking out the samples.  I am definitely open to any further feedback suggestions you have around this.

**Comment by Mike-EEE** [comment]
OK so the coffee has kicked in here @nblumhardt ☕️ hehe.  It turns out after implementing this, that I do not need to decorate the `ILogEventSink` at all, but rather the `ITextFormatter` of sinks that use this.  

For the `Trace` extension method, it uses the `MessageTemplateTextFormatter` so I am able to decorate this just fine.  The problem lies squarely in the `OutputTemplateRenderer` which is the default `ITextFormatter` that the `Console` extension method uses.  

The initial PR included the `CreateNew.TextFormatter` method.  I have removed `CreateNew.ConsoleSink` and have retained `CreateNew.TextFormatter` instead for now.

As such, we could simply make `OutputTemplateRenderer` public which would not only remove  `CreateNew` altogether, but also make it consistent with `MessageTemplateTextFormatter`.  However, there might be a design consideration here, and I wanted to get your take on this before doing so.

**Comment by Mike-EEE** [comment]
Indeed, @nblumhardt.  While going through some further design and thought around this, it was my revelation that everything can be done via the `LoggerSinkConfiguration.Wrap` as that provides access to the `LogEvent` to intercept.  Intercepting at the `ITextFormatter` level does not guarantee accurate state representation of the object as it could be running on a different thread/context by the time it gets to formatting it.

In any case, good exercise to learn more about serilog and of course PRs are always a challenge.  Plus, I just learned what ELI5 is, LOL!  Thank you again for your time and patience here. 👍
